proto/api/v1/api.proto

```diff
@@ -799,3 +800,61 @@ option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
     }
   }
 };
+
+service PluginService {
+  rpc ListPlugins(ListPluginsRequest) returns (ListPluginsResponse) {
+    option (google.api.http) = {
+      get: "/v1/plugins"
+      response_body: "plugins"
+    };
+  };
+  rpc GetPlugin(GetPluginRequest) returns (GetPluginResponse) {
+    option (google.api.http) = {
+      get: "/v1/plugin/{name}"
+      response_body: "plugin"
+    };
+  };
+}
+
+message ListPluginsRequest {
+}
+
+message ListPluginsResponse{
+  repeated PluginSpecifications plugins = 1;
+}
+
+message GetPluginRequest {
+  string name = 1;
+}
+
+message GetPluginResponse {
+  PluginSpecifications plugin = 1;
+}
+
+message PluginSpecifications{
+  enum Type {
+    TYPE_UNSPECIFIED = 0;
+    TYPE_STRING = 1;
+    TYPE_INT = 2;
+    TYPE_FLOAT = 3;
+    TYPE_BOOL = 4;
+    TYPE_ENUM = 5;
+  }
+
+  message Parameter {
+    string default = 1;
+    bool required = 2;
+    string description = 3;
+    Type type = 4;
+    repeated string enum = 5; // specify the enum expected values
```

Comment:
any other suggestions for this param name? enum-values or values maybe?

Reply:
What's this enum about btw?

Reply:
It's meant to hold the enum values in case `type` is `TYPE_ENUM`. This way the UI could display a dropdown instead of a freeform field, a great example would be the [acks](https://github.com/ConduitIO/conduit-plugin-kafka/blob/361df379fa98cd9cfcda439120e9a0752f6230b4/config.go#L31) option in the kafka connector, which could define the allowed fields `none`, `one` and `all`.

That said, I've been thinking of another possible approach - defining an enum could be compared with setting up a validation. To define an enum the field would still have the type `TYPE_STRING` but a validation would be added on top which defines what are the allowed values. If we have a way to add validations to a field it would add more flexibility, the UI and Conduit could validate the config (to some degree) before it even gets to the plugin.

While it adds flexibility, it also adds complexity. The question is if this is something we want or not.